webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/users/UserRepository.scala

```diff
@@ -0,0 +1,834 @@
+                .using(StatusHistory(UserStatus.Deleted, trace) :: _)
+            )
+          } else {
+            (k, u)
+          }
+      }
+      (modUserIds, m)
+    }
+  }
+
+  override def purge(
+      userIds:           List[String],
+      deletedSince:      Option[DateTime],
+      excludeFromOrigin: List[String],
+      trace:             EventTrace
+  ): IOResult[List[String]] = {
+
+    userBase.modify { users =>
+      val initIds = users.keySet
+      val m       = users.filterNot {
+        case (_, u) =>
+          (predicateUser(u, userIds) || predicateDeletedSince(u, deletedSince)) && predicateNotOrigin(u, excludeFromOrigin)
+      }
+      ((initIds -- m.keySet).toList, m)
+    }
+  }
+
+  override def setActive(userId: List[String], trace: EventTrace): IOResult[Unit] = {
+    userBase.update {
+      _.map {
+        case (k, u) =>
+          if (userId.contains(u.id) && u.status == UserStatus.Disabled) {
+            (
+              k,
+              u.modify(_.status)
+                .setTo(UserStatus.Active)
+                .modify(_.statusHistory)
+                .using(StatusHistory(UserStatus.Active, trace) :: _)
+            )
+          } else {
+            (k, u)
+          }
+      }
+    }
+  }
+
+  override def get(userId: String): IOResult[Option[UserInfo]] = {
+    userBase.get.map(_.get(userId))
+  }
+
+  override def updateInfo(
+      id:        String,
+      name:      Option[Option[String]],
+      email:     Option[Option[String]],
+      otherInfo: Option[Json.Obj]
+  ): IOResult[Unit] = {
+    userBase.update(users => {
+      users.get(id) match {
+        case None    => users
+        case Some(u) =>
+          users + (id -> u
+            .modify(_.name)
+            .setToIfDefined(name)
+            .modify(_.email)
+            .setToIfDefined(email)
+            .modify(_.otherInfo)
+            .setToIfDefined(otherInfo))
+      }
+    })
+  }
+}
+
+/*
+ * Postgresql storage of users. There is two tables, one for user, one for session.
+ * Users:
+ * id             text PRIMARY KEY NOT NULL CHECK (id <> '')
+ * creationDate   timestamp with time zone NOT NULL
+ * status         text NOT NULL
+ * managedBy   text NOT NULL CHECK (managedBy <> '')
+ * name           text
+ * email          text
+ * lastLogin      timestamp with time zone
+ * statusHistory  jsonb
+ * otherInfo      jsonb -- general additional user info
+ *
+ * UserSessions:
+ *   userId       text NOT NULL CHECK (nodeId <> '')
+ *   creationDate timestamp with time zone NOT NULL
+ *   authMethod   text
+ *   permissions  text[]
+ *   endDate      timestamp with time zone
+ *   endCause     text
+ */
+class JdbcUserRepository(doobie: Doobie) extends UserRepository {
+  import com.normation.rudder.db.Doobie.DateTimeMeta
+  import com.normation.rudder.db.json.implicits._
+  import com.normation.rudder.users.UserSerialization._
+  import doobie._
+
+  implicit val logger = Doobie.slf4jDoobieLogger
```

Comment:
no, that one is not important: what we want to have is a nice upmerge experience. For that, we want to limit changes between branches, and so want want to have the same changes on all branches (for ex adding types). 
But here, since the code was removed in upper branch, adding the type would likely cause a merge conflict, while letting it unchanged is ok (the deletion was already merged).